samtranslator/model/api/api_generator.py

```diff
@@ -274,9 +274,13 @@ def _add_auth(self):
         authorizers = self._get_authorizers(auth_properties.Authorizers, auth_properties.DefaultAuthorizer)
 
         if authorizers:
-            swagger_editor.add_authorizers(authorizers)
+            swagger_editor.add_authorizers_security_definitions(authorizers)
             self._set_default_authorizer(swagger_editor, authorizers, auth_properties.DefaultAuthorizer)
 
+        if auth_properties.ApiKeyRequired:
+            swagger_editor.add_apikey_security_definition()
+            self._set_default_apikey_required(swagger_editor)
+
         # Assign the Swagger back to template
         self.definition_body = swagger_editor.swagger
```

Comment:
Could you please rebase and add/update a test to ensure this works with openapi 3 as well.

Reply:
@praneetap , I have rebased.  I am happy to add a test for openapi 3.  Can you provide a little direction on that?  Are there other tests I can refer to as an example?

Reply:
@praneetap , I've added a couple of openapi3 translator tests.  Please let me know if there's anything else you'd like to see on this.

New Test:
- api_with_apikey_required_openapi_3

Updated Tests:
- api_with_auth_all_maximum_openapi_3
- api_with_auth_all_maximum

Reply:
Yes, will be happy to! OpenApi tests are run [here](https://github.com/awslabs/serverless-application-model/blob/develop/tests/translator/test_translator.py#L345).
You can find more examples of input templates [here](https://github.com/awslabs/serverless-application-model/tree/develop/tests/translator/input) if you search for "openapi".